### PR TITLE
[JENKINS-70394] Move 'set node temporarily offline/online' buttons to app-bar

### PR DESCRIPTION
**Pull request by mPokornyETM**
As describe in JENKINS-70394 are the to set the node temp. online/offline not clickable, because the offline-cause warning covers this buttons.

See [JENKINS-70394](https://issues.jenkins.io/browse/JENKINS-70394).

### Testing done

No Java changes, therefore no new automatic tests.
The UI looks now like this:
image
image
image

### Proposed changelog entries

- Move 'set node temporarily offline/online' buttons to app-bar

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- ~~[ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.~~
- ~~[ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.~~
- ~~[ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).~~
- ~~[ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.~~
- ~~[ ] For new APIs and extension points, there is a link to at least one consumer.~~

### Desired reviewers

@jenkinsci/core-pr-reviewers 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7577"></a>

### CODE REVIEW

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (uhafner) [inline core/src/main/resources/hudson/model/Computer/index.jelly:31]: Shouldn't there be a title?

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly:31]: yes the agent heading should be moved into the title (Built-in Node etc)

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:31]: And where shall be placed the 'status' icon? Or is there some way to place the icon in the app-bar-title too?
I try it before, and I can add the icon only as item (like the buttons) and in that case it is aligned to right side, with looks very weird.

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly:31]: Status icons are not recommended to be in the heading so fine to remove as part of this:

https://weekly.ci.jenkins.io/design-library/Symbols/

> Don't add symbols to headings, they just create visual clutter

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:31]: somthing like this mnight be a good compromis
image

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly]: thanks can you update the screenshots in the PR description? I'll try test it out later on my only worry which needs checking is that other agent types will have defined the title in the `index-top.jelly`

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly]: Screen shots are up-to-date.

**Comment by mPokornyETM** [comment]
The jenkins job for [commit](https://github.com/jenkinsci/jenkins/pull/7577/commits/378840b3ab7af66a945141d31a8373dab8d7a289) is stuck. Can somebody kill him, and replay it? It looks like a issue on jenkins-environment side.

**Comment by timja** [comment]
> The jenkins job for [commit](https://github.com/jenkinsci/jenkins/pull/7577/commits/378840b3ab7af66a945141d31a8373dab8d7a289) is stuck. Can somebody kill him, and replay it? It looks like a issue on jenkins-environment side.

@MarkEWaite restarted 6 mins ago by the looks of it 😄

**Comment by timja** [comment]
The screenshot shows 2 titles for built-in node. it should only be there once

**Comment by timja** [comment]
I think `Launch agent` should also move to the app bar for consistency:

image

I created an agent that's started by SSH and put a bad host in it so it would fail to start up.

----

Once the title and launch agent are fixed I would be happy to approve

**Comment by mPokornyETM** [comment]
> The screenshot shows 2 titles for built-in node. it should only be there once
> 
> Verified locally and it's an issue: image

nice catch ;-)

**Comment by mPokornyETM** [comment]
> I think `Launch agent` should also move to the app bar for consistency:
> 
> image
> 
> I created an agent that's started by SSH and put a bad host in it so it would fail to start up.
> 
> Once the title and launch agent are fixed I would be happy to approve

I thing it will be not possible. It loads 'some' jelly page which depends on slave-type. And this might be from type to type other implemented and broke the view as well.
Ex.
ssh-agent looks pretty good
image

But the permanent agent is broken
image

Of cores I can fix the permanent agent jelly file as well, but no body can grant, which agents are implemented around all plugins.

**Comment by timja** [comment]
What about creating a new jelly include for app bar controls?

**Comment by mPokornyETM** [comment]
> What about creating a new jelly include for app bar controls?

I tries that too, but it raise some exception. It looks, that the app bar works only in main page. I am not sure. But it will still not helps. Because we need handle all slave-types. Maybe we shall create follow issue and discuss it there?

**Comment by uhafner** [comment]
Maybe it would be safer to use the fixed title "Status" then and let the actual title appear in the normal content div as it was before. 

From a UX perspective the current page has a lot of additional problems which should be fixed in an additional PR. (Buttons should not contain full sentences, sometimes the word Agent is used - sometimes the word Node. Why do we use three pages for Status, Delete and Configure when one would be sufficient)

**Comment by mPokornyETM** [comment]
> 

The open thing is not about the status.  It is about the launch button.  But I agree with you.  This page (or generally node pages) shall be refactored. Also/computer 
page has potential to be better.  But in that case somebody shall define, how it shall look. And I can try to implement it afterwards. I created some PRs in this direction before, but it was more  or le

**Comment by mPokornyETM** [comment]
Sort for closing/ reopening. I was too fast on my cell phone

**Comment by timja** [comment]
@mPokornyETM I pushed a fix with what I meant can you take a look and see if you are happy with it please?

**Comment by mPokornyETM** [comment]
> @mPokornyETM I pushed a fix with what I meant can you take a look and see if you are happy with it please?

Hm, thinking, testing, trying to break something and is still looks good.


One last more thing: I think this might cause some inconsistency, when some Slave from some one plugin allow manual launch.
In this case you will see the manual launch button still on the same position. Because the plugin does not have (probably) app-bar-controls.jelly So this one will be not loaded. Maybe we shall somehow inform plugin developers about this change. Or we just wait until somebody raise first issue.

Anyway, thx for co-contributing and co-operations. Changes LGTM. All together looks good, works as expected and definitely fix related issue.

**Comment by timja** [comment]
> One last more thing: I think this might cause some inconsistency, when some Slave from some one plugin allow manual launch.

You got an example?

**Comment by mPokornyETM** [comment]
> > One last more thing: I think this might cause some inconsistency, when some Slave from some one plugin allow manual launch.
> 
> You got an example?

No, I just thinking. For some reason must be this implemented as is.
https://github.com/jenkinsci/jenkins/blob/799157879a029c2897ea20bd18eb008260b6a9e7/core/src/main/resources/hudson/model/Computer/index.jelly
```
<j:if test="${it.manualLaunchAllowed}">
  <st:include from="${it.launcher}" page="main.jelly" optional="true"/>
</j:if>
```

**Comment by timja** [comment]
Would have to check some custom launchers to see if any plugins add buttons there, would be fairly easy to adapt.

Maybe some cloud plugins like ec2 might add them there.

There _should_ be less launchers than agent types as hopefully the built-in ones get mostly reused.

**Comment by timja** [comment]
see https://www.jenkins.io/doc/developer/extensions/jenkins-core/#computerlauncher

Need to check for path `<>Launcher/main.jelly`

**Comment by mPokornyETM** [comment]
> see https://www.jenkins.io/doc/developer/extensions/jenkins-core/#computerlauncher
> 
> Need to check for path `<>Launcher/main.jelly`

Check all listed extension will be a challenge.  There are many. I can start tomorrow (hopefully) let me know which one has you checked, to eliminate effort.

**Comment by timja** [comment]
I checked these earlier:
image

**Comment by timja** [comment]
I checked most of the plugins and couldn't find any outside of core that have a main.jelly with anything more than text and most don't even have a `main.jelly` and just use what's in core

**Review by timja — APPROVED** [review]

**Comment by mPokornyETM** [comment]
> I checked most of the plugins and couldn't find any outside of core that have a main.jelly with anything more than text and most don't even have a `main.jelly` and just use what's in core

Thx. That means I can continue on lockable-resources-plugin again.  There is a lot of work open.

**Review by NotMyFault — APPROVED** [review]
Thanks for taking care of this! ATH is green, and I was able to verify that changing the online state of nodes works again :)

**Comment by daniel-beck** [comment]
Are we confident that there are no other occurrences where `<div class="warning">` and similar mess with other UI elements in this manner? Seems like fixing that would be a far more preferable solution than moving buttons around to work around that UI element's peculiar behavior. Even if there is no other current occurrence of the problem, is the suggested solution for that always going to be to move buttons out of the way?

**Review by Kevin-CB — APPROVED** [review]
LGTM from a security standpoint

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by benipeled** [comment]
@mPokornyETM thanks for the fix,
The button is now clickable but less visible because of the light background

image

It's not the case with the `Bring this node back online` button 

image

**Comment by mPokornyETM** [comment]
> @mPokornyETM thanks for the fix, The button is now clickable but less visible because of the light background
> 
> image
> 
> It's not the case with the `Bring this node back online` button
> 
> image

Thx for the feedback. This shall be addressed to UI components. Because it is general 'issue' on app-bar. I do not use here any css-styles. Or better to say, no more.

**Comment by timja** [comment]
> It's not the case with the Bring this node back online button

this is a primary action, e.g. something to call attention to.

`Mark this node temporarily offline` is not the primary reason someone is likely to be on that page, that's an uncommon action and hence it's not attention drawing.

Styling could be improved for some better contrast but as Martin said that's something separate 😄

**Pull request merged**
